bundles/binding/org.openhab.binding.mystromecopower/src/main/java/org/openhab/binding/mystromecopower/internal/MyStromEcoPowerGenericBindingProvider.java

```diff
@@ -0,0 +1,131 @@
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author Jordens Christophe
+ * @since 1.7.0-SNAPSHOT
+ */
+/**
+ * @author Christophe
+ *
+ */
+public class MyStromEcoPowerGenericBindingProvider extends AbstractGenericBindingProvider implements MyStromEcoPowerBindingProvider {
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "mystromecopower";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		if (!(item instanceof SwitchItem ||
+				item instanceof NumberItem ||
+				item instanceof StringItem)) {
+			throw new BindingConfigParseException("item '" + item.getName()
+					+ "' is of type '" + item.getClass().getSimpleName()
+					+ "', only Switch, Number or String is allowed - please check your *.items configuration");
+		}
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item, String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
+		MyStromEcoPowerBindingConfig config = new MyStromEcoPowerBindingConfig();
+		
+		//parse bindingconfig here ...	
+		config.mystromFriendlyName = bindingConfig;
+		config.isSwitch = item instanceof SwitchItem;
+		config.isNumberItem = item instanceof NumberItem;
+		config.isStringItem = item instanceof StringItem;
+		addBindingConfig(item, config);		
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.MyStromEcoPowerBindingProvider#getMystromFriendlyName(java.lang.String)
+	 */
+	public String getMystromFriendlyName(String itemName) {
+		MyStromEcoPowerBindingConfig config = (MyStromEcoPowerBindingConfig) bindingConfigs.get(itemName);
+		return config != null ? config.mystromFriendlyName : null;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Boolean getIsSwitch(String itemName) {
+		MyStromEcoPowerBindingConfig config = (MyStromEcoPowerBindingConfig) bindingConfigs.get(itemName);
+		return config != null ? config.isSwitch : false;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Boolean getIsNumberItem(String itemName) {
+		MyStromEcoPowerBindingConfig config = (MyStromEcoPowerBindingConfig) bindingConfigs.get(itemName);
+		return config != null ? config.isNumberItem : false;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Boolean getIsStringItem(String itemName) {
+		MyStromEcoPowerBindingConfig config = (MyStromEcoPowerBindingConfig) bindingConfigs.get(itemName);
+		return config != null ? config.isStringItem : false;
+	}
+	
+	/**
+	 * This class is responsible to transport configuration data.
+	 * 
+	 * @author Jordens Christophe
+	 * @since 1.7.0-SNAPSHOT
+	 */
+	class MyStromEcoPowerBindingConfig implements BindingConfig {
```

Comment:
Extract MyStromEcoPowerBindingConfig into its own file if it is not required to be an inner class. This helps discoverability for humans :smile: